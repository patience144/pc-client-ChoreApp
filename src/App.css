html, body, #root, main {
  box-sizing: border-box;
  height: 100%;
}