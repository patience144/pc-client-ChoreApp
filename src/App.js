import React from "react";
import { Route, withRouter, Switch } from "react-router-dom";
import Nav from "./Nav";
import Welcome from "./Welcome";
import ChoreList from "./ChoreList";
import AddChore from "./AddChore";
import EditChore from "./EditChore";
import ChoresContext from "./context";
import api from "./api";
import "./style.css";
import "./App.css";

class App extends React.Component {
  state = { chores: [] };

  static contextType = ChoresContext;

  updateChores = (chores) => {
    let newState = this.state.chores;
    newState.chores = chores;
    this.setState({ chores });
  };

  render() {
    return (
      <main className="App">
        <Nav />
        <ChoresContext.Provider
          value={{
            chores: this.state.chores,
            updateChores: this.updateChores,
          }}
        >
          <Switch>
            <Route path={"/"} exact component={Welcome} />
            <Route path={"/about"} component={Welcome} />
            <Route
              path={"/chores"}
              render={() => <ChoreList chores={this.state.chores} />}
            />
            <Route
              path={"/add"}
              render={() => (
                <AddChore
                  chores={this.state.chores}
                  updateChores={this.updateChores}
                />
              )}
            />
            <Route
              path={"/edit/:choreID"}
              render={() => (
                <EditChore
                  chores={this.state.chores}
                  updateChores={this.updateChores}
                />
              )}
            />
          </Switch>
        </ChoresContext.Provider>
      </main>
    );
  }

  componentDidMount() {
    return api
      .getChores()
      .then((chores) => {
        this.setState({ chores });
      })
      .catch((error) => console.log({ error }));
  };
};

export default withRouter(App);
