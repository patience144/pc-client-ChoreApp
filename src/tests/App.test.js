import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import App from '../App';

describe('The App', () => {
  const app = <BrowserRouter><App /></BrowserRouter>;
  describe('Smoke Test', () => {
    it('renders learn react link', () => {
      render(app);
    });
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(app, div);
      ReactDOM.unmountComponentAtNode(div);
    });
  });
  describe('Snapshot Test', () => {
    it('renders the UI as expected', () => {
      const tree = renderer.create(app).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});


