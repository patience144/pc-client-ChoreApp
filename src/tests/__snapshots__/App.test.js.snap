// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App Snapshot Test renders the UI as expected 1`] = `
<main
  className="App"
>
  <div
    className="nav"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1>
        Chore App
      </h1>
    </a>
    <ul
      className="nav-list"
    >
      <li>
        <a
          href="/about"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
            }
          }
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/chores"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
            }
          }
        >
          Chore List
        </a>
      </li>
      <li>
        <a
          href="/add"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
            }
          }
        >
          Add Chores
        </a>
      </li>
    </ul>
  </div>
  <div
    className="welcome"
  >
    <h2>
      Let's make Chores Easier!!!
    </h2>
    <h3>
      Welcome to the Chore App A place where you can create a list of your chores and work throug them digitally! You can add a chore, delete a chore, or simply check it off for the day! Ready to begin?
    </h3>
    <a
      href="/add"
      onClick={[Function]}
    >
      Click Here to Start!
    </a>
  </div>
</main>
`;
